lib/events/auditlog.go

```diff
@@ -1179,3 +1192,46 @@ func (l *LegacyHandler) IsUnpacked(ctx context.Context, sessionID session.ID) (b
 func (l *LegacyHandler) Download(ctx context.Context, sessionID session.ID, writer io.WriterAt) error {
 	return l.cfg.Handler.Download(ctx, sessionID, writer)
 }
+
+type nopLog struct {
+}
+
+func (a *nopLog) EmitAuditEventLegacy(e Event, f EventFields) error {
+	return trace.NotImplemented("not implemented")
```

Comment:
This would result in all calls to nop logger return errors, right? Shouldn't no-op logger just `return nil` everywhere instead of errors?

Reply:
I can do that too, felt safer to this because tests actually pass and this highlights that a NOP log is actually being used.

Reply:
rename it to `alwaysFailLog` or something more descriptive

Reply:
or better, `closedLogger` that returns `trace.NotImplemented("the logger has been closed")`

Reply:
Yeah, in that case I think we should rename it from nop logger to something else.